### PR TITLE
Makefile: tiny polish the `make ut` command

### LINKED ISSUE #32243: Run`make ut` error, too many open files
## Problem1

make ut

```
building task finish, count=4783, takes=36.943982531s
run all tasks takes 2m38.904517081s
open temp cover file error: open /tmp/cov1281420580/executor.TestQueryTime: too many open files
make: *** [Makefile:131: gotest_in_verify_ci] Error 1
```

`ulimit -n`  result is 1024, but there are 4000+ temporary files...

## Problem2

After running `make ut`, too many *.test.bin file left, takes more than 5G disk size:

```
find . -name "*.bin" | xargs ls -l | awk '{ total += $5 }; END { print "total size: ",total }'
total size:  5943474416
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #32243

Problem Summary:

### What is changed and how it works?

1. clean up the generate test.bin files after test
2. close the temporary coverage profile file after handling its data, to avoid the "too many open files" error

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tools/check/ut.go [hunk 1]
```diff
@@ -401,20 +401,7 @@ func collectCoverProfileFile() {
 		if file.IsDir() {
 			continue
 		}
-
-		f, err := os.Open(path.Join(coverFileTempDir, file.Name()))
-		if err != nil {
-			fmt.Println("open temp cover file error:", err)
-			os.Exit(-1)
-		}
-		defer f.Close()
```
Comment (tiancaiamao) [inline tools/check/ut.go:410]: The old code `defer` in a for-loop, so the file is not closed immediately.

File: Makefile [hunk 2]
```diff
@@ -115,9 +115,12 @@ explaintest: server_check
 ddltest:
 	@cd cmd/ddltest && $(GO) test -o ../../bin/ddltest -c
 
+CLEAN_UT_BINARY := find . -name '*.test.bin'| xargs rm -rf
```
Comment (bb7133) [inline Makefile:118]:
```suggestion
CLEAN_UT_BINARY := find . -name '*.test.bin'| xargs rm
```
IMHO `rm` should be enough...`rm -rf` is a little scary in a script.

Comment (tiancaiamao) [reply Makefile]: Done ... 
I never 'rm -rf /'
:lol

**Review by hawkingrei — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**